api/accounts/views.py

```diff
@@ -105,23 +110,47 @@ class SocialMediaTypeDetail(generics.RetrieveUpdateDestroyAPIView):
 
 def sign_s3(request):
 
-  file_name = request.GET['folder'] + "/" + uuid.uuid1().hex
-  file_type = request.GET['file_type']
-
-  s3 = boto3.client('s3')
-
-  presigned_post = s3.generate_presigned_post(
-    Bucket = settings.S3_BUCKET,
-    Key = file_name,
-    Fields = {"acl": "public-read", "Content-Type": file_type},
-    Conditions = [
-      {"acl": "public-read"},
-      {"Content-Type": file_type}
-    ],
-    ExpiresIn = 3600
-  )
-
-  return JsonResponse({
-    'data': presigned_post,
-    'url': 'https://%s.s3.amazonaws.com/%s' % (settings.S3_BUCKET, file_name)
-  })
+    file_name = request.GET['folder'] + "/" + uuid.uuid1().hex
+    file_type = request.GET['file_type']
+
+    s3 = boto3.client('s3')
+
+    presigned_post = s3.generate_presigned_post(
+        Bucket=settings.S3_BUCKET,
+        Key=file_name,
+        Fields={"acl": "public-read", "Content-Type": file_type},
+        Conditions=[
+            {"acl": "public-read"},
+            {"Content-Type": file_type}
+        ],
+        ExpiresIn=3600
+    )
+
+    return JsonResponse({
+        'data': presigned_post,
+        'url': 'https://%s.s3.amazonaws.com/%s' % (settings.S3_BUCKET, file_name)
+    })
+
+
+class PasswordChange(generics.GenericAPIView):
+
+    def post(self, request, *args, **kwargs):
+        data = request.data.copy()
+        print(data)
+        email = data.get('email')
+        current_password = data.get('currentPassword')
+        new_password = data.get('newPassword')
+        user = authenticate(username=email, password=current_password)
+
+        if user is not None:
+            user.set_password(new_password)
+            user.save()
+            print(f'Success! New password is {new_password}')
+            return Response(
+                 f'Success! New password is {new_password}'
+            )
```

Comment:
Just send a response - don't send the new password

Reply:
updated response